### DISCUSSION TITLE
Surface Normals from Different Poses

### DISCUSSION THREAD

**Discussion by iulsko** [pyvista/pyvista#1680] [Q&A] [answered]
Hi! 😊

I would like to access the surface normals of the mesh, like `mesh.compute_normals(inplace=True)` but after modifying the pose of the mesh relative to the camera. I am stuck:

 - I can't change the pose of the mesh (there is no `mesh.set_pose`) and re-calculate the surface normals
 - I can't extract the surface normals from the plot using `pyvista.Plotter()` after `mesh.plot(cpos=pose)`

I would like to compute the surface normals in each camera view. Is it possible to do that?

**Comment by banesullivan** [comment, marked as answer]
Surface normals are NOT calculated relative to any pose of the mesh or camera position. The normals of the mesh that PyVista will compute for you are the literal surface normals of the mesh. Perhaps this visual helps a bit:

```py
import pyvista as pv
box = pv.Box()
box.compute_normals(inplace=True)

centers = box.cell_centers().points
arrows = box['Normals']

p1 = pv.Plotter()
p1.add_mesh(box, color='tan')
p1.add_arrows(centers, arrows, color='black')
p1.show()
```
<img width="624" alt="Screen Shot 2021-11-26 at 12 14 34 AM" src="https://user-images.githubusercontent.com/22067021/143541582-47b54ac8-58e8-4752-8295-0bcdab212cc6.png">

---- 

So, correct me if I don't have this right, but I think all you want to do is compute the vector between the cell centers and a given camera position? If so, this would simply be the following after using a plotter to get a desired camera position:

```py
... # use plotter to find desired camera position

pos = p1.camera_position[0]
vecs = pos - centers
vecs /= np.linalg.norm(vecs)

p2 = pv.Plotter(notebook=0)
p2.add_mesh(box, color='tan')
p2.add_arrows(centers, vecs, color='black', mag=5)
p2.add_points(np.array([pos,]), color='red', 
              point_size=30,
              render_points_as_spheres=True)
p2.show()
```


<img width="624" alt="Screen Shot 2021-11-26 at 12 27 10 AM" src="https://user-images.githubusercontent.com/22067021/143542958-8244fac0-82f3-471c-a351-21d70b21adf8.png">
